addons/pos_loyalty/static/src/js/ControlButtons/PromoCodeButton.js

```diff
@@ -12,12 +12,16 @@ export class PromoCodeButton extends PosComponent {
     }
 
     async onClick() {
-        const { confirmed, payload: code } = await this.showPopup('TextInputPopup', {
-            title: this.env._t('Enter Promotion or Coupon Code'),
+        let { confirmed, payload: code } = await this.showPopup('TextInputPopup', {
+            title: this.env._t('Enter Code'),
             startingValue: '',
+            placeholder: this.env._t('Gift card or Discount code'),
         });
-        if (confirmed && code !== '') {
-            this.env.pos.get_order().activateCode(code);
+        if (confirmed) {
+            code = code.trim();
+            if (code !== '') {
+                this.env.pos.get_order().activateCode(code);
```

Comment:
Wait, it's possible that a gift card is sold in sales. We should allow using those.

Reply:
https://github.com/odoo/odoo/pull/99559/commits/c4ac0a56093c660a0bfd8fd3a3fb152eb135c7cc

Instead of forbidding, I'm giving the user an option to allow applying the gift card. I feel like it's not nice to be fully restrictive.